src/adapters/morpho-aave-v2/common/morphoBasePoolAdapter.ts

```diff
@@ -0,0 +1,633 @@
+          }
+        }),
+    )
+    return protocolTokens
+  }
+
+  async getWithdrawals({
+    userAddress,
+    protocolTokenAddress,
+    fromBlock,
+    toBlock,
+  }: GetEventsInput): Promise<MovementsByBlock[]> {
+    return await this.getMovements({
+      protocolToken: await this.fetchProtocolTokenMetadata(
+        protocolTokenAddress,
+      ),
+      underlyingTokens: await this.fetchUnderlyingTokensMetadata(
+        protocolTokenAddress,
+      ),
+      filter: {
+        smartContractAddress: protocolTokenAddress,
+        fromBlock,
+        toBlock,
+        from: userAddress,
+        to: undefined,
+      },
+    })
+  }
+
+  async getDeposits({
+    userAddress,
+    protocolTokenAddress,
+    fromBlock,
+    toBlock,
+  }: GetEventsInput): Promise<MovementsByBlock[]> {
+    return await this.getMovements({
+      protocolToken: await this.fetchProtocolTokenMetadata(
+        protocolTokenAddress,
+      ),
+      underlyingTokens: await this.fetchUnderlyingTokensMetadata(
+        protocolTokenAddress,
+      ),
+      filter: {
+        smartContractAddress: protocolTokenAddress,
+        fromBlock,
+        toBlock,
+        from: undefined,
+        to: userAddress,
+      },
+    })
+  }
+
+  async getTotalValueLocked({
+    blockNumber,
+  }: GetTotalValueLockedInput): Promise<ProtocolTokenTvl[]> {
+    const tokens = await this.getProtocolTokens()
+    const lensContract = MorphoAaveV2Lens__factory.connect(
+      this.lensAddress,
+      this.provider,
+    )
+    const positionType = this.getProtocolDetails().positionType
+    return Promise.all(
+      tokens.map(async (tokenMetadata) => {
+        let totalValueRaw
+
+        if (positionType === PositionType.Supply) {
+          const [poolSupply, p2pSupply] =
+            await lensContract.getTotalMarketSupply(tokenMetadata.address, {
+              blockTag: blockNumber,
+            })
+          totalValueRaw = poolSupply + p2pSupply
+        } else {
+          // Assuming LensType.Borrow or other types
+          const [poolBorrow, p2pBorrow] =
+            await lensContract.getTotalMarketBorrow(tokenMetadata.address, {
+              blockTag: blockNumber,
+            })
+          totalValueRaw = poolBorrow + p2pBorrow
+        }
+
+        return {
+          ...tokenMetadata,
+          type: TokenType.Protocol,
+          totalSupplyRaw:
+            totalValueRaw !== undefined ? totalValueRaw : BigInt(0),
+        }
+      }),
+    )
+  }
+
+  protected async getUnderlyingTokenConversionRate(
+    protocolTokenMetadata: Erc20Metadata,
+    _blockNumber?: number | undefined,
+  ): Promise<UnderlyingTokenRate[]> {
+    const { underlyingToken } = await this.fetchPoolMetadata(
+      protocolTokenMetadata.address,
+    )
+
+    // Aave tokens always pegged one to one to underlying
+    const PRICE_PEGGED_TO_ONE = 1
+    const pricePerShareRaw = BigInt(
+      PRICE_PEGGED_TO_ONE * 10 ** protocolTokenMetadata.decimals,
+    )
```

Comment:
The description is a bit missleading, this is more because `balanceOf` of `aTokens` is already scaled with the exchange rate